swap/src/protocol/alice.rs

```diff
@@ -175,30 +175,20 @@ impl Builder {
         btc_to_swap: bitcoin::Amount,
         xmr_to_swap: monero::Amount,
     ) -> Result<AliceState> {
-        let rng = &mut OsRng;
-
         let amounts = SwapAmounts {
             btc: btc_to_swap,
             xmr: xmr_to_swap,
         };
 
-        let a = bitcoin::SecretKey::new_random(rng);
-        let s_a = cross_curve_dleq::Scalar::random(rng);
-        let v_a = monero::PrivateViewKey::new_random(rng);
-        let redeem_address = self.bitcoin_wallet.new_address().await?;
-        let punish_address = redeem_address.clone();
```

Comment:
The reason why (Just added that to the commit, will push soon) is because the `EventLoop` is now handling the instantiation of `State0` using `State0::new` so I just wanted to avoid having to instantiate too many variable to use this function within the event loop. In short, I just want the API to be easier to use so I can then use it :)